<launch>

  <arg name="port" value="/dev/ttyUSB0" />
  <arg name="q_name" value="hummingbird_1" />
  <arg name="controller" value="asctec_so3_control" />
  <arg name="q_frame" value="/vicon/hummingbird_1/hummingbird_1" />
  <arg name="w_frame" value="/odom" />
	<arg name="rviz" value="basic.rviz"/>
	
  <include file="$(find vicon_bridge)/launch/vicon.launch" />
  
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find asctec_viz)/rviz/$(arg rviz)" />
	<node pkg="asctec_viz"
		type="asctec_viz_node"
		name="asctec_viz"
		output="screen">
		<param name="topic_name" value="$(arg controller)"/>
	</node>

  <node pkg="asctec_so3"
    type="asctec_control_node"
    name="$(arg controller)"
    required="true"
    output="screen">
    <param name="mass" value="0.55"/>
    <rosparam param="gains/pos">[3, 3, 8.2]</rosparam>
    <rosparam param="gains/vel">[3, 3, 4.5]</rosparam>
		<rosparam param="gains/yaw">0.8</rosparam>
  </node>

  <node pkg="trajectory_gen"
    type="min_accel_node"
    name="min_accel"
    required="true"
    output="screen">
    <param name="rate" value="40"/>
    <param name="topic_name" value="$(arg controller)"/>
  </node>

  <node pkg="asctec_ll_updated" name="$(arg q_name)" type="asctec_ll_updated_node" output="screen" respawn="true">
     <param name="port" value="$(arg port)"/>
     <param name="ll_status_rate" value="1"/>
     <param name="imu_calc_data_rate" value="0"/>
     <param name="rc_data_rate" value="0"/>
     <param name="ctrl_out_rate" value="10"/>
     <param name="gps_data_advanced_rate" value="0"/>
     <remap from="~cmd_si" to="/asctec_so3_control/si_command" />
  </node>

  <node pkg="asctec_so3"
    type="asctec_vicon_node"
    name="asctec_vicon"
    required="true"
    output="screen">
    <param name="q_name" value="$(arg controller)"/>
    <param name="q_frame" value="$(arg q_frame)"/>
    <param name="w_frame" value="$(arg w_frame)"/>
  </node>

	<node pkg="asctec_so3"
		type="square_node"
		name="suareqy"
		output="screen">
		<param name="topic" value="$(arg controller)"/>
	</node>

</launch>
